### PR TITLE
lightning: check counter value to make code more robust

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #37338

Problem Summary:

### What is changed and how it works?
cannot reproduce this case in local env, so add this check to make code more robust, instead of panic

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/restore/restore.go [hunk 1]
```diff
@@ -2330,7 +2330,16 @@ func (cr *chunkRestore) deliverLoop(
 		cr.chunk.Chunk.PrevRowIDMax = rowID
 
 		if m, ok := metric.FromContext(ctx); ok {
-			m.BytesCounter.WithLabelValues(metric.BytesStateRestored).Add(float64(currOffset - startOffset))
+			// value of currOffset comes from parser.pos which increase monotonically. the init value of parser.pos
+			// comes from chunk.Chunk.Offset. so it should happen that currOffset - startOffset < 0.
```
Comment (buchuitoudegou) [inline br/pkg/lightning/restore/restore.go:2334]:
```suggestion
			// comes from chunk.Chunk.Offset. so it shouldn't happen that currOffset - startOffset < 0.
```
typo?

**Review by buchuitoudegou — APPROVED** [review]

**Review by hi-rustin — APPROVED** [review]

**Comment by D3Hunter** [comment]
ci success https://ci.pingcap.net/blue/organizations/jenkins/br_ghpr_unit_and_integration_test/detail/br_ghpr_unit_and_integration_test/13031/pipeline

**Pull request merged**